plugins/handlers/gallery.py

```diff
@@ -351,56 +434,40 @@ def metadata_for_media_file(root, path, title_from_filename):
 
 def process_image(incontext, root, destination, dirname, basename, category, title_from_filename=True):
     source_path = os.path.join(root, dirname, basename)
-    identifier = generate_identifier(basename)
-    destination_dir = os.path.join(destination, dirname)
-    destination_path = os.path.join(destination_dir, basename)
-    markdown_path = os.path.join(destination_dir, "%s.markdown" % identifier)
-    name, ext = os.path.splitext(basename)
-    utils.makedirs(destination_dir)
 
     metadata = metadata_for_media_file(root, os.path.join(dirname, basename),
                                        title_from_filename=title_from_filename)
 
-    # TODO: Support specifying image size sets in the configuration file #10
-
-    # Determine which profiles to use; we use a different profile for equirectangular projections.
-    profiles = DEFAULT_PROFILES
-    if "projection" in metadata and metadata["projection"] == "equirectangular":
-        profiles = EQUIRECTANGULAR_PROFILES
-
-    # Generate the various different image sizes.
-    for profile_name, profile_details in profiles.items():
-
-        filename = "%s-%s%s" % (identifier, profile_name.replace("_", "-"), ext)
-        if profile_name == "image":
-            filename = "%s%s" % (identifier, ext)
-
-        scale = 1
-        if "scale" in profile_details:
-            scale = profile_details["scale"]
-        width = None
-        if "width" in profile_details:
-            width = profile_details["width"]
-        height = None
-        if "height" in profile_details:
-            height = profile_details["height"]
-
-        source_scale = metadata["scale"] if metadata["scale"] is not None else 1
-
-        # TODO: Rename generate_thumbnail.
-        metadata[profile_name] = generate_thumbnail(source_path,
-                                                    destination, dirname, filename,
-                                                    (width, height),
-                                                    source_scale, scale)  # TODO: Why two scales?
+    name, _ = os.path.splitext(basename)
+    name = name.lower()
+    destination_path = os.path.join(destination, dirname, name)
+    utils.makedirs(destination_path)
+    transform_files = []
+    transform_metadata = collections.defaultdict(list)
+    transforms = evaluate_tests(IMAGE_TRANSFORMS, {"basename": os.path.basename(source_path),
+                                                   "metadata": metadata})
+    for transform in transforms:
+        result = transform.perform(path=source_path, destination=destination_path)
+        if not isinstance(result, TransformResult):
+            raise AssertionError("Invalid result type.")
```

Comment:
It's quite possible that this doesn't actually need to return a `TransformResult` and could instead return just the paths of the files generated.